### ISSUE TITLE
Chat User Details is missing Email/Phone number leads to confusion & issues

### ISSUE THREAD

**Issue by Santhosh-Sellavel** [Expensify/App#4904]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
Need to have two user accounts to reproduce this, with the following details.
1. One User Account with both phone & email linked let's call `UserME`
2. Another user account - Let's call `User2` (If this user has mobile & email it's much easier to reproduce)

## Action Performed:
1. Go to New Expenisy with `User2`
2. Open the `UserME's` chat window, 
3. Click the chat header and view user details.
### Issue 1
4. Email/phone is missing based on` UserME's` First login. Example: I signed up using mobile-first, so email was missing in my case.
### Issue 2
5. Go to search, enter` UserME's` secondary login i.e that's email in my case. 
6. User name is not showing at all, showing in a result like a different user.
7. After clicking the from search correct chat window is loaded, but email details are not available on the details page and no sign that whether the user is navigated to the correct chat because of missing details.

### Issue 1
Should have shown both email & mobile number.
### Issue 2
Should show user name in search results for matching logins.

### Issue 1
Either phone/email missing 
#### Actual User Profile:
<img width="379" alt="Profile Image" src="https://user-images.githubusercontent.com/85645967/131336725-c018752b-54dd-42ff-894b-b0976c45f244.png">
#### Details page
<img width="394" alt="Details Page" src="https://user-images.githubusercontent.com/85645967/131336713-0c9f6b99-c3ec-45e6-a885-3590d6cd6eea.png">

### Issue 2 (All searches)
The search should have shown the user name in results, but not showing.

https://user-images.githubusercontent.com/85645967/131336931-b97443a4-6601-4c0a-a34a-6ef1afbd53bd.mov

## Workaround:
Can the user still use Expensify without this being fixed? Single user login is no problem.
But user with multiple logins and then its a mystery - we can't ensure the shown is correct because of either phone/email where the user only knows `UserME's` secondary login. There is a vast no of possible cases where the searching users may know either phone or email.

## Suggestion && Proposal
I see issue needs have both client & backend requirements. After triage and discussion will propose a technical solution for client changes.

## Platform:

Where is this issue occurring? All

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @sonialiap (`AutoAssignerTriage`), see https://stackoverflow.com/c/expensify/questions/4749 for more details.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @alex-mechler (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by alex-mechler** [comment]
Issue 1 can be solved completely on the client.

Issue 2 _might_ be able to be solved on the client, I'm not sure however.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @arielgreen (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by Santhosh-Sellavel** [comment]
@alex-mechler 
We don't share both logins in personal props? For the user I am chating with.

Without that we can't solve Issue 1.

**Comment by Santhosh-Sellavel** [comment]
I'll check again update you shortly.

**Comment by Santhosh-Sellavel** [comment]
@alex-mechler 
<img width="306" alt="Screenshot 2021-08-31 at 11 57 31 PM" src="https://user-images.githubusercontent.com/85645967/131556603-3474a21a-aee8-498d-958d-eecf1f879f7e.png">

Note: I'm not logged in as `Mobile user`. I logged in as another user trying to view this user profile.

**Comment by alex-mechler** [comment]
Is the email for `Mobile User` validated?

**Comment by Santhosh-Sellavel** [comment]
Yes

**Comment by alex-mechler** [comment]
Ahhh I see. I'm not sure if we expose the secondary logins of an account to other users. For now I've assigned back to myself while I do a little more digging. Thanks for the details so far!

**Comment by Santhosh-Sellavel** [comment]
@alex-mechler 
Okay, additionally I checked in API response also. It's not available. 

<img width="728" alt="Screenshot 2021-09-01 at 12 20 19 AM" src="https://user-images.githubusercontent.com/85645967/131559467-dfe0a2bb-01d1-4be0-b5d6-70305d5418dc.png">

**Comment by MelvinBot** [comment]
@alex-mechler Huh... This is 4 days overdue. Who can take care of this?

**Comment by mallenexpensify** [comment]
@Santhosh-Sellavel , @alex-mechler is out til tomorrow.  I bumped to weekly since this likely isn't a 'quick fix' type issue. 
[This Slack post ](https://expensify.slack.com/archives/C01GTK53T8Q/p1630857084456400)is also related to second logins. 

Alex, at a high level, do issues involving secondary logins need to be worked on internally?

**Comment by mallenexpensify** [comment]
Also... I did this test today which I imagine will be common for users. 
- Test account requests money from my personal/secondary login email. 
- Once request is sent, my work/primary email shows and not my personal email
-

**Comment by alex-mechler** [comment]
Thanks for bumping that @mallenexpensify!

Based on [the thread where we discussed](https://expensify.slack.com/archives/C01GTK53T8Q/p1630519530290200) this issue, this doesn't seem to be high priority. We did not come up with use cases where this is critical to have, so I will be closing out this issue for now. Its possible we do this in the future, its just not a priority now. 

---

> Alex, at a high level, do issues involving secondary logins need to be worked on internally?

It depends on an issue to issue basis. For this issue, we do not already send the secondary logins of other users to the client, so this would need to be at least partially internal. Some issues may be external, if the data needed is already on the client.
